packages/edit-post/src/components/header/index.js

```diff
@@ -118,47 +106,19 @@ function Header( { setEntitiesSavedStatesCallback, initialPost } ) {
 				className="edit-post-header__toolbar"
 			>
 				<DocumentTools disableBlockTools={ isTextEditor } />
-				{ showTopToolbar && (
-					<>
-						<div
-							className={ classnames(
-								'selected-block-tools-wrapper',
-								{
-									'is-collapsed':
-										isBlockToolsCollapsed ||
-										! hasBlockSelection,
```

Comment:
Having this element inline meant it was rerendering even when the top toolbar option was off. Moving it into a new component `<ContextualToolbar />` means it won't be rerendered unless the top toolbar is on.